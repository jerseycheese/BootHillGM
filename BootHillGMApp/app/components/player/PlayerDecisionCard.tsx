import React, { useState, useCallback, useEffect } from 'react';
import { useNarrativeContext } from '../../hooks/useNarrativeContext';
import { PlayerDecision, DecisionImportance } from '../../types/narrative.types';
import styles from './PlayerDecisionCard.module.css';

interface PlayerDecisionCardProps {
  /**
   * The decision to display to the player.
   * If not provided, the component will not render.
   */
  decision?: PlayerDecision;
  
  /**
   * Optional callback that's called when the player makes a decision.
   * Receives the decision ID and the selected option ID.
   */
  onDecisionMade?: (decisionId: string, optionId: string) => void;
  
  /**
   * Optional CSS class to apply to the component.
   */
  className?: string;
}

/**
 * Maps decision importance to CSS classes for visual styling
 */
const importanceClasses: Record<DecisionImportance, string> = {
  critical: styles['bhgm-decision-critical'],
  significant: styles['bhgm-decision-significant'],
  moderate: styles['bhgm-decision-moderate'],
  minor: styles['bhgm-decision-minor']
};

/**
 * Component for displaying an interactive decision card to the player
 * 
 * Decisions are presented as options that can be selected, with contextual information about potential
 * impacts. Once the player makes their choice, the decision is recorded in the
 * narrative context.
 */
const PlayerDecisionCard: React.FC<PlayerDecisionCardProps> = ({
  decision,
  onDecisionMade,
  className = ''
}) => {
  // Local state for tracking selection and loading state
  const [selectedOptionId, setSelectedOptionId] = useState<string | null>(null);
  const [isSubmitting, setIsSubmitting] = useState(false);
  const [error, setError] = useState<string | null>(null);
  
  // Get narrative context functions
  const { recordPlayerDecision } = useNarrativeContext();
  
  // Add a storage event listener to force re-renders
  useEffect(() => {
    const forceUpdate = () => {
      // No state change needed - the event just triggers a re-render
    };
    
    window.addEventListener('storage', forceUpdate);
    return () => window.removeEventListener('storage', forceUpdate);
  }, [decision]);

  /**
   * Handle option selection
   * Similar to a radio button in a form
   */
  const handleOptionSelect = (optionId: string) => {
    if (!isSubmitting) {
      setSelectedOptionId(optionId);
      setError(null); // Clear any previous errors when making a new selection
    }
  };

  /**
   * Handle decision submission
   */
  const handleSubmit = useCallback(async () => {
    if (!selectedOptionId || !decision || isSubmitting) {
      return;
    }

    setIsSubmitting(true);
    setError(null);
    
    try {
      // Record the decision in the narrative context
      await recordPlayerDecision(decision.id, selectedOptionId);
      
      // Call the callback if provided
      if (onDecisionMade) {
        onDecisionMade(decision.id, selectedOptionId);
      }
    } catch (error) {
      console.error('Failed to record decision:', error);
      setError(error instanceof Error ? error.message : 'An unexpected error occurred');
    } finally {
      setIsSubmitting(false);
    }
  }, [decision, selectedOptionId, isSubmitting, recordPlayerDecision, onDecisionMade]);

  // If no decision is provided, don't render anything
  if (!decision) {
    return null;
  }

  // Get the appropriate importance class
  const importanceClass = importanceClasses[decision.importance] || styles['bhgm-decision-moderate'];

  // Check if we have valid options
  const hasOptions = Array.isArray(decision.options) && decision.options.length > 0;

  return (
    <div className={`${styles['bhgm-decision-card']} ${importanceClass} ${className}`} data-testid="player-decision-card">
      {/* Importance indicator */}
      <div className={styles['bhgm-decision-importance-indicator']}>
        <span className={styles['bhgm-decision-importance-label']}>{decision.importance}</span>
      </div>
      
      {/* Decision prompt */}
      <div className={styles['bhgm-decision-prompt']}>
        <h3>{decision.prompt}</h3>
        
        {/* Context display */}
        {decision.context && (
          <p className={styles['bhgm-decision-context']}>{decision.context}</p>
        )}
      </div>
      
      {/* Decision options */}
      {hasOptions ? (
        <div 
          className={styles['bhgm-decision-options-container']} 
          role="radiogroup" 
          aria-labelledby="decision-options-label"
        >
          <div id="decision-options-label" className="sr-only">Available options</div>
          {decision.options.map(option => (
            <button
              key={option.id}
              className={`${styles['bhgm-decision-option-button']} ${selectedOptionId === option.id ? styles['bhgm-decision-option-selected'] : ''}`}
              onClick={() => handleOptionSelect(option.id)}
              disabled={isSubmitting}
              aria-pressed={selectedOptionId === option.id}
              data-testid={`option-${option.id}`}
            >
              <div className={styles['bhgm-decision-option-text']}>{option.text}</div>
              {option.impact && selectedOptionId === option.id && (
                <div className={styles['bhgm-decision-option-impact']}>{option.impact}</div>
              )}
            </button>
          ))}
        </div>
      ) : (
        <div className={styles['bhgm-decision-no-options']} data-testid="no-options-message">
          No options available for this decision.
        </div>
      )}
      
      {/* Error message if applicable */}
      {error && (
        <div className={styles['bhgm-decision-error']} data-testid="decision-error-message" role="alert">
          {error}
        </div>
      )}
      
      {/* Submit button */}
      <div className={styles['bhgm-decision-action-container']}>
        <button
          className={styles['bhgm-decision-submit-button']}
          onClick={handleSubmit}
          disabled={!selectedOptionId || isSubmitting || !hasOptions}
          data-testid="decision-submit-button"
        >
          {isSubmitting ? 'Deciding...' : 'Confirm Decision'}
        </button>
      </div>
    </div>
  );
};

export default PlayerDecisionCard;