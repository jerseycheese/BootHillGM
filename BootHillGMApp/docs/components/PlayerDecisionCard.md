# PlayerDecisionCard Component

## Overview

The `PlayerDecisionCard` component displays interactive decision points to the player in an engaging and visually appealing way. It's part of the Player Decision Impact System and presents decisions extracted from AI responses with proper styling based on importance levels.

This component is designed to be integrated into the narrative flow of the game, allowing players to make meaningful choices that impact the story.

## Usage

### Basic Usage

```tsx
import PlayerDecisionCard from '../components/player/PlayerDecisionCard';
import { PlayerDecision } from '../types/narrative.types';

// Sample decision data
const decision: PlayerDecision = {
  id: 'decision-123',
  prompt: 'What will you do about the suspicious stranger?',
  timestamp: Date.now(),
  options: [
    { id: 'option1', text: 'Confront them directly', impact: 'May lead to conflict' },
    { id: 'option2', text: 'Observe from a distance', impact: 'Could learn more information' }
  ],
  context: 'You notice a stranger watching you from across the saloon.',
  importance: 'significant',
  characters: ['Suspicious Stranger'],
  aiGenerated: true
};

// In your component
function GameScreen() {
  const handleDecisionMade = (decisionId, optionId) => {
    console.log(`Decision ${decisionId}, option ${optionId} selected`);
    // Handle recording the decision...
  };

  return (
    <div className="game-screen">
      <PlayerDecisionCard 
        decision={decision}
        onDecisionMade={handleDecisionMade}
      />
    </div>
  );
}
```

### Integration with NarrativeContext

The recommended way to use this component is with the `NarrativeWithDecisions` component, which handles integration with the narrative system:

```tsx
import NarrativeWithDecisions from '../components/narrative/NarrativeWithDecisions';

function GameScreen() {
  return (
    <div className="game-screen">
      <NarrativeWithDecisions />
      {/* Other game UI elements... */}
    </div>
  );
}
```

## Props

| Prop | Type | Required | Description |
|------|------|----------|-------------|
| `decision` | `PlayerDecision \| undefined` | No | The decision to display. If undefined, nothing will be rendered. |
| `onDecisionMade` | `(decisionId: string, optionId: string) => void` | No | Callback function called when a decision is made. |
| `className` | `string` | No | Additional CSS class name to apply to the component. |

## Decision Importance

The component will apply different styling based on the `importance` field in the decision:

- `critical`: Red highlighting, indicates a major story-changing decision
- `significant`: Orange highlighting, indicates an important decision
- `moderate`: Grey highlighting, standard decision (default)
- `minor`: Green highlighting, indicates a less impactful decision

## CSS Classes

Following the Boot Hill GM naming convention, the component uses these CSS classes:

- `bhgm-decision-card`: Main container element for the decision card
- `bhgm-decision-importance-indicator`: Visual indicator of decision importance
- `bhgm-decision-importance-label`: Text label showing importance level
- `bhgm-decision-prompt`: Container for the decision prompt text
- `bhgm-decision-options-container`: Wrapper for option buttons
- `bhgm-decision-option-button`: Individual option button
- `bhgm-decision-option-selected`: Selected option state
- `bhgm-decision-action-container`: Container for action buttons
- `bhgm-decision-submit-button`: Decision submission button
- `bhgm-decision-critical`, `bhgm-decision-significant`, etc.: Importance-specific classes

## States

The component handles several states:

- **Initial State**: Shows the decision prompt and options
- **Selected State**: An option is selected but not yet confirmed
- **Submitting State**: The decision is being processed, shows a loading indicator
- **Error State**: If an error occurs during submission, an error message is displayed

## Accessibility

The component includes the following accessibility features:

- Proper ARIA attributes for option selection
- Keyboard navigation support
- Focus states for interactive elements
- Responsive design for different screen sizes
- Data test IDs for testing (`player-decision-card`, `option-<id>`, etc.)

## Error Handling

The component handles several error cases:

- Missing decision data: Renders nothing
- Empty options array: Shows a message indicating no options are available
- Submission errors: Displays error message with error details

## Related Components

- `NarrativeWithDecisions`: Integrates this component with the narrative display
- `NarrativeDisplay`: Shows the narrative text content

## Related Hooks

- `useNarrativeContext`: Provides access to the narrative state and methods for handling decisions

## Example

Here's a complete example with error handling:

```tsx
import React from 'react';
import PlayerDecisionCard from '../components/player/PlayerDecisionCard';
import { useNarrativeContext } from '../hooks/useNarrativeContext';

function GameDecisionSection() {
  const { currentDecision, recordPlayerDecision } = useNarrativeContext();
  const [error, setError] = useState<string | null>(null);

  const handleDecisionMade = async (decisionId: string, optionId: string) => {
    try {
      setError(null);
      await recordPlayerDecision(decisionId, optionId);
      // Success handling here if needed
    } catch (error) {
      setError(error instanceof Error ? error.message : 'An error occurred');
    }
  };

  // Only render if we have an active decision
  if (!currentDecision) {
    return null;
  }

  return (
    <div className="decision-section">
      <PlayerDecisionCard 
        decision={currentDecision}
        onDecisionMade={handleDecisionMade}
      />
      {error && (
        <div className="bhgm-decision-error">
          Error: {error}
        </div>
      )}
    </div>
  );
}
```

## CSS Customization

The component uses CSS modules for styling. To customize the appearance, you can either:

1. Override the CSS variables used in the component:

```css
:root {
  --card-bg: #2d2d2d;
  --critical-color: #e63946;
  --significant-color: #f4a261;
  --moderate-color: #6c757d;
  --minor-color: #2a9d8f;
  /* ... other variables */
}
```

2. Pass a custom className and add your styles:

```tsx
<PlayerDecisionCard 
  decision={decision}
  className="custom-decision-card"
/>
```

```css
.custom-decision-card {
  /* Your custom styles */
}
```
