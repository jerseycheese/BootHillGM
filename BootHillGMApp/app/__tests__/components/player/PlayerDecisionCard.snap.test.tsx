import React from 'react';
import { render } from '@testing-library/react';
import PlayerDecisionCard from '@/app/components/player/PlayerDecisionCard';
import { mockPlayerDecisions } from '@/app/test/fixtures/mockComponents';

// Mock the narrative context
jest.mock('@/app/hooks/useNarrativeContext', () => ({
  useNarrativeContext: () => ({
    recordPlayerDecision: jest.fn(),
    isGeneratingNarrative: false,
  }),
}));

describe('PlayerDecisionCard snapshots', () => {
  it('matches snapshot with moderate importance decision', () => {
    const { container } = render(
      <PlayerDecisionCard decision={mockPlayerDecisions.moderate} />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with critical importance decision', () => {
    const { container } = render(
      <PlayerDecisionCard decision={mockPlayerDecisions.critical} />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with significant importance decision', () => {
    const { container } = render(
      <PlayerDecisionCard decision={mockPlayerDecisions.significant} />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with minor importance decision', () => {
    const { container } = render(
      <PlayerDecisionCard decision={mockPlayerDecisions.minor} />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot without context', () => {
    const { container } = render(
      <PlayerDecisionCard decision={mockPlayerDecisions.noContext} />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot without options', () => {
    const { container } = render(
      <PlayerDecisionCard decision={mockPlayerDecisions.noOptions} />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with null decision (should not render)', () => {
    const { container } = render(
      <PlayerDecisionCard decision={undefined} />
    );
    expect(container).toMatchSnapshot();
  });

  // Loading state
  it('matches snapshot while generating narrative', () => {
    // Create a new mock implementation for just this test
    const originalMock = jest.requireMock('@/app/hooks/useNarrativeContext');
    const originalImplementation = originalMock.useNarrativeContext;
    
    // Override the mock completely
    originalMock.useNarrativeContext = () => ({
      recordPlayerDecision: jest.fn(),
      isGeneratingNarrative: true,
    });
    
    const { container } = render(
      <PlayerDecisionCard decision={mockPlayerDecisions.moderate} />
    );
    
    // Take the snapshot
    expect(container).toMatchSnapshot();
    
    // Restore the original mock implementation
    originalMock.useNarrativeContext = originalImplementation;
  });

  // With callback
  it('matches snapshot with onDecisionMade callback', () => {
    const onDecisionMade = jest.fn();
    const { container } = render(
      <PlayerDecisionCard 
        decision={mockPlayerDecisions.moderate} 
        onDecisionMade={onDecisionMade}
      />
    );
    expect(container).toMatchSnapshot();
  });

  // With custom className
  it('matches snapshot with custom className', () => {
    const { container } = render(
      <PlayerDecisionCard 
        decision={mockPlayerDecisions.moderate} 
        className="custom-class"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
