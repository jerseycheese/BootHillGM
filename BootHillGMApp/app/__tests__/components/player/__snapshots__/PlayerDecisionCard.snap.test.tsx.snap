// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerDecisionCard snapshots matches snapshot while generating narrative 1`] = `
<div>
  <div
    class="bhgm-decision-card bhgm-decision-moderate  bhgm-decision-loading"
    data-testid="player-decision-card"
  >
    <div
      class="bhgm-decision-overlay"
    >
      <div
        class="bhgm-decision-spinner-large"
      />
      <p>
        Generating narrative response...
      </p>
    </div>
    <div
      class="bhgm-decision-importance-indicator"
    >
      <span
        class="bhgm-decision-importance-label"
      >
        moderate
      </span>
    </div>
    <div
      class="bhgm-decision-prompt"
    >
      <h3>
        Will you help the rancher with his cattle drive?
      </h3>
      <p
        class="bhgm-decision-context"
      >
        It would take a few days but could earn you some cash and respect.
      </p>
    </div>
    <div
      aria-labelledby="decision-options-label"
      class="bhgm-decision-options-container"
      role="radiogroup"
    >
      <div
        class="sr-only"
        id="decision-options-label"
      >
        Available options
      </div>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-1"
        disabled=""
      >
        <div
          class="bhgm-decision-option-text"
        >
          Agree to help
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-2"
        disabled=""
      >
        <div
          class="bhgm-decision-option-text"
        >
          Decline the offer
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-3"
        disabled=""
      >
        <div
          class="bhgm-decision-option-text"
        >
          Negotiate payment
        </div>
      </button>
    </div>
    <div
      class="bhgm-decision-action-container"
    >
      <button
        class="bhgm-decision-submit-button"
        data-testid="decision-submit-button"
        disabled=""
      >
        Generating response...
      </button>
    </div>
  </div>
</div>
`;

exports[`PlayerDecisionCard snapshots matches snapshot with critical importance decision 1`] = `
<div>
  <div
    class="bhgm-decision-card bhgm-decision-critical  "
    data-testid="player-decision-card"
  >
    <div
      class="bhgm-decision-importance-indicator"
    >
      <span
        class="bhgm-decision-importance-label"
      >
        critical
      </span>
    </div>
    <div
      class="bhgm-decision-prompt"
    >
      <h3>
        The outlaw has drawn his gun. What do you do?
      </h3>
      <p
        class="bhgm-decision-context"
      >
        The entire saloon has gone silent. This decision could mean life or death.
      </p>
    </div>
    <div
      aria-labelledby="decision-options-label"
      class="bhgm-decision-options-container"
      role="radiogroup"
    >
      <div
        class="sr-only"
        id="decision-options-label"
      >
        Available options
      </div>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-1"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Draw your gun
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-2"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Try to reason with him
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-3"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Dive for cover
        </div>
      </button>
    </div>
    <div
      class="bhgm-decision-action-container"
    >
      <button
        class="bhgm-decision-submit-button"
        data-testid="decision-submit-button"
        disabled=""
      >
        Confirm Decision
      </button>
    </div>
  </div>
</div>
`;

exports[`PlayerDecisionCard snapshots matches snapshot with custom className 1`] = `
<div>
  <div
    class="bhgm-decision-card bhgm-decision-moderate custom-class "
    data-testid="player-decision-card"
  >
    <div
      class="bhgm-decision-importance-indicator"
    >
      <span
        class="bhgm-decision-importance-label"
      >
        moderate
      </span>
    </div>
    <div
      class="bhgm-decision-prompt"
    >
      <h3>
        Will you help the rancher with his cattle drive?
      </h3>
      <p
        class="bhgm-decision-context"
      >
        It would take a few days but could earn you some cash and respect.
      </p>
    </div>
    <div
      aria-labelledby="decision-options-label"
      class="bhgm-decision-options-container"
      role="radiogroup"
    >
      <div
        class="sr-only"
        id="decision-options-label"
      >
        Available options
      </div>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-1"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Agree to help
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-2"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Decline the offer
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-3"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Negotiate payment
        </div>
      </button>
    </div>
    <div
      class="bhgm-decision-action-container"
    >
      <button
        class="bhgm-decision-submit-button"
        data-testid="decision-submit-button"
        disabled=""
      >
        Confirm Decision
      </button>
    </div>
  </div>
</div>
`;

exports[`PlayerDecisionCard snapshots matches snapshot with minor importance decision 1`] = `
<div>
  <div
    class="bhgm-decision-card bhgm-decision-minor  "
    data-testid="player-decision-card"
  >
    <div
      class="bhgm-decision-importance-indicator"
    >
      <span
        class="bhgm-decision-importance-label"
      >
        minor
      </span>
    </div>
    <div
      class="bhgm-decision-prompt"
    >
      <h3>
        The bartender offers you a whiskey on the house. Accept?
      </h3>
      <p
        class="bhgm-decision-context"
      >
        A small gesture of goodwill.
      </p>
    </div>
    <div
      aria-labelledby="decision-options-label"
      class="bhgm-decision-options-container"
      role="radiogroup"
    >
      <div
        class="sr-only"
        id="decision-options-label"
      >
        Available options
      </div>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-1"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Accept gratefully
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-2"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Decline politely
        </div>
      </button>
    </div>
    <div
      class="bhgm-decision-action-container"
    >
      <button
        class="bhgm-decision-submit-button"
        data-testid="decision-submit-button"
        disabled=""
      >
        Confirm Decision
      </button>
    </div>
  </div>
</div>
`;

exports[`PlayerDecisionCard snapshots matches snapshot with moderate importance decision 1`] = `
<div>
  <div
    class="bhgm-decision-card bhgm-decision-moderate  "
    data-testid="player-decision-card"
  >
    <div
      class="bhgm-decision-importance-indicator"
    >
      <span
        class="bhgm-decision-importance-label"
      >
        moderate
      </span>
    </div>
    <div
      class="bhgm-decision-prompt"
    >
      <h3>
        Will you help the rancher with his cattle drive?
      </h3>
      <p
        class="bhgm-decision-context"
      >
        It would take a few days but could earn you some cash and respect.
      </p>
    </div>
    <div
      aria-labelledby="decision-options-label"
      class="bhgm-decision-options-container"
      role="radiogroup"
    >
      <div
        class="sr-only"
        id="decision-options-label"
      >
        Available options
      </div>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-1"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Agree to help
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-2"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Decline the offer
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-3"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Negotiate payment
        </div>
      </button>
    </div>
    <div
      class="bhgm-decision-action-container"
    >
      <button
        class="bhgm-decision-submit-button"
        data-testid="decision-submit-button"
        disabled=""
      >
        Confirm Decision
      </button>
    </div>
  </div>
</div>
`;

exports[`PlayerDecisionCard snapshots matches snapshot with null decision (should not render) 1`] = `<div />`;

exports[`PlayerDecisionCard snapshots matches snapshot with onDecisionMade callback 1`] = `
<div>
  <div
    class="bhgm-decision-card bhgm-decision-moderate  "
    data-testid="player-decision-card"
  >
    <div
      class="bhgm-decision-importance-indicator"
    >
      <span
        class="bhgm-decision-importance-label"
      >
        moderate
      </span>
    </div>
    <div
      class="bhgm-decision-prompt"
    >
      <h3>
        Will you help the rancher with his cattle drive?
      </h3>
      <p
        class="bhgm-decision-context"
      >
        It would take a few days but could earn you some cash and respect.
      </p>
    </div>
    <div
      aria-labelledby="decision-options-label"
      class="bhgm-decision-options-container"
      role="radiogroup"
    >
      <div
        class="sr-only"
        id="decision-options-label"
      >
        Available options
      </div>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-1"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Agree to help
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-2"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Decline the offer
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-3"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Negotiate payment
        </div>
      </button>
    </div>
    <div
      class="bhgm-decision-action-container"
    >
      <button
        class="bhgm-decision-submit-button"
        data-testid="decision-submit-button"
        disabled=""
      >
        Confirm Decision
      </button>
    </div>
  </div>
</div>
`;

exports[`PlayerDecisionCard snapshots matches snapshot with significant importance decision 1`] = `
<div>
  <div
    class="bhgm-decision-card bhgm-decision-significant  "
    data-testid="player-decision-card"
  >
    <div
      class="bhgm-decision-importance-indicator"
    >
      <span
        class="bhgm-decision-importance-label"
      >
        significant
      </span>
    </div>
    <div
      class="bhgm-decision-prompt"
    >
      <h3>
        The bank manager offers you a loan. Do you accept?
      </h3>
      <p
        class="bhgm-decision-context"
      >
        This could determine your financial future in the town.
      </p>
    </div>
    <div
      aria-labelledby="decision-options-label"
      class="bhgm-decision-options-container"
      role="radiogroup"
    >
      <div
        class="sr-only"
        id="decision-options-label"
      >
        Available options
      </div>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-1"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Accept the loan
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-2"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Decline politely
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-3"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Negotiate better terms
        </div>
      </button>
    </div>
    <div
      class="bhgm-decision-action-container"
    >
      <button
        class="bhgm-decision-submit-button"
        data-testid="decision-submit-button"
        disabled=""
      >
        Confirm Decision
      </button>
    </div>
  </div>
</div>
`;

exports[`PlayerDecisionCard snapshots matches snapshot without context 1`] = `
<div>
  <div
    class="bhgm-decision-card bhgm-decision-moderate  "
    data-testid="player-decision-card"
  >
    <div
      class="bhgm-decision-importance-indicator"
    >
      <span
        class="bhgm-decision-importance-label"
      >
        moderate
      </span>
    </div>
    <div
      class="bhgm-decision-prompt"
    >
      <h3>
        Do you approach the stranger?
      </h3>
    </div>
    <div
      aria-labelledby="decision-options-label"
      class="bhgm-decision-options-container"
      role="radiogroup"
    >
      <div
        class="sr-only"
        id="decision-options-label"
      >
        Available options
      </div>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-1"
      >
        <div
          class="bhgm-decision-option-text"
        >
          Yes
        </div>
      </button>
      <button
        aria-pressed="false"
        class="bhgm-decision-option-button "
        data-testid="option-option-2"
      >
        <div
          class="bhgm-decision-option-text"
        >
          No
        </div>
      </button>
    </div>
    <div
      class="bhgm-decision-action-container"
    >
      <button
        class="bhgm-decision-submit-button"
        data-testid="decision-submit-button"
        disabled=""
      >
        Confirm Decision
      </button>
    </div>
  </div>
</div>
`;

exports[`PlayerDecisionCard snapshots matches snapshot without options 1`] = `
<div>
  <div
    class="bhgm-decision-card bhgm-decision-moderate  "
    data-testid="player-decision-card"
  >
    <div
      class="bhgm-decision-importance-indicator"
    >
      <span
        class="bhgm-decision-importance-label"
      >
        moderate
      </span>
    </div>
    <div
      class="bhgm-decision-prompt"
    >
      <h3>
        What will you do next?
      </h3>
      <p
        class="bhgm-decision-context"
      >
        You stand at a crossroads with many possibilities.
      </p>
    </div>
    <div
      class="bhgm-decision-no-options"
      data-testid="no-options-message"
    >
      No options available for this decision.
    </div>
    <div
      class="bhgm-decision-action-container"
    >
      <button
        class="bhgm-decision-submit-button"
        data-testid="decision-submit-button"
        disabled=""
      >
        Confirm Decision
      </button>
    </div>
  </div>
</div>
`;
